lib/appium.js

```diff
@@ -250,9 +261,12 @@ class AppiumDriver extends BaseDriver {
     try {
       if (this.sessions[sessionId]) {
         const curConstructorName = this.sessions[sessionId].constructor.name;
-        const otherSessionsData = _.toPairs(this.sessions)
+        let otherSessionsData;
+        await sessionDeletionLock.acquire(AppiumDriver.name, async () => {
```

Comment:
same question here. what could possibly cause a race condition here? certainly not a bunch of lodash methods?